internal/pkg/cli/env_upgrade.go

```diff
@@ -266,6 +266,69 @@ func (o *envUpgradeOpts) isManagerRoleAllowedToUpload(body string) (bool, error)
 	return true, nil
 }
 
+func (o *envUpgradeOpts) grantManagerRolePermissionToUpload(cfn environmentDeployer, app, env, execRole, body, bucketARN string) error {
+	// Detect which line number the EnvironmentManagerRole's PolicyDocument Statement is at.
+	// We will add additional permissions after that line.
+	type Template struct {
+		Resources struct {
+			ManagerRole struct {
+				Properties struct {
+					Policies []struct {
+						Document struct {
+							Statements yaml.Node `yaml:"Statement"`
+						} `yaml:"PolicyDocument"`
+					} `yaml:"Policies"`
+				} `yaml:"Properties"`
+			} `yaml:"EnvironmentManagerRole"`
+		} `yaml:"Resources"`
+	}
+
+	var tpl Template
+	if err := yaml.Unmarshal([]byte(body), &tpl); err != nil {
+		return fmt.Errorf("unmarshal environment template to find EnvironmentManagerRole policy statement: %v", err)
```

Comment:
```suggestion
		return fmt.Errorf("unmarshal environment template to find EnvironmentManagerRole policy statement: %w", err)
```

Reply:
Kept it as is, it's not really important 🤷  but I did that on purpose since the `cli` package is the root pkg so we don't need to surface the error part of the api